client/src/modules/vouchers/modals/search.modal.js

```diff
@@ -74,8 +98,21 @@ function VoucherRegistrySearchModalController(ModalInstance, filters, Notify, mo
 
     // push all searchQuery values into the changes array to be applied
     angular.forEach(vm.searchQueries, function (value, key) {
-      if (angular.isDefined(value)) {
-        changes.post({ key : key, value : value });
+      if (angular.isDefined(value)) {        
```

Comment:
This is pretty complicated code.

Why don't you track display values in a separate object.  It doesn't even need to be tied to the view - something like this:
```js
// searchQueries is the same id:value pair
vm.searchQueries = {};
// displayValues will be an id:displayValue pair
var displayValues = {};
```
... then, each individual method can define how to display it's display value.  For example, for `user_id`, you can just do this:
```js
// search on user.id, but show user.display_name in the filters
vm.onSelectUser = function onSelectUser(user) {
  vm.searchQueries.user_id = user.id;
  displayValues.user_id = user.display_name;
};
```
... and then in the submit method ...
```js
angular.forEach(vm.searchQueries, function (value, key) {
  if (angular.isDefined(value)) {
    // default to the original value if no display value is defined
    var displayValue = displayValues[key] || value;
    changes.post({ key: key, value: value, displayValue: displayValue });
   }
});
```